IceCreamShop.playground/Contents.swift

```diff
@@ -1 +1,91 @@
+import UIKit
 
+struct Flavor {
+    var name: String
+    var rating: Double
+}
+
+enum Size: Double {
+    case small = 1.00
+    case medium = 2.88
+    case large = 5.25
+}
+
+struct Cone {
+    let flavor: Flavor
+    let topping: String?
+    let size: Size
+    
+    func eat() {
+        print("Yum, I love \(flavor.name)!)")
+    }
+}
+
+class IceCreamShop {
+    var flavors: [Flavor]
+    var sizes: [Size]
+    var toppings: [String?]
+    var totalSales: Double = 0.0
+
+    init(flavors: [Flavor], sizes: [Size], toppings: [String?]) {
+        self.flavors = flavors
+        self.sizes = sizes
+        self.toppings = toppings
+    }
+//    func listTopFlavors() {
+//        var topFlavors: [String] = []
+//
+//        for flavor in flavors {
+//            if flavor.rating >= 4.0 {
+//                topFlavors.append(flavor.name)
+//                }
+//            print("Our top flavors are \(flavor.name)")
+//            }
+//        }
+        
+    func listTopFlavors(flavors: [Flavor]) {
+
+        var topFlavors: [String] = []
+
+        for flavor in flavors {
+            if flavor.rating >= 4.0 {
+                topFlavors.append(flavor.name)
+                print("Our top flavors are \(flavor.name)!")
+            }
+        }
+    }
```

Comment:
Doing it like this, the top flavors are all printed on separate lines. Maybe consider how you would build the string up in the for loop, and then print it all at once after the loop. A simple solution that wouldn't deal with the fact that a trailing comma would be added to the last flavor might look like this:

``` swift
func listTopFlavors(flavors: [Flavor]) {
    
    var topFlavors: [String] = []
    
    for flavor in flavors {
        if flavor.rating >= 4.0 {
            topFlavors.append(flavor.name)
        }
    }
    
    var topFlavorsString = "Our top flavors are "
    
    for flavorString in topFlavors {
        topFlavorsString += "\(flavorString), "
    }
    
    print(topFlavorsString)
}
```